lib/classes/PluginManager.js

```diff
@@ -148,7 +152,27 @@ class PluginManager {
     if (pluginsObject.localPath) {
       module.paths.unshift(pluginsObject.localPath);
     }
-    this.loadPlugins(pluginsObject.modules);
+    this.loadPlugins(pluginsObject.modules
+      .filter(name => name !== '@serverless/enterprise-plugin'));
+  }
+
+  loadEnterprisePlugin() {
+    const qualifiedEnterpriseModulePath = path.join(
+      __dirname, '../../node_modules/@serverless/enterprise-plugin');
+    // eslint-disable-next-line global-require
+    const sfePkgJson = require(`${qualifiedEnterpriseModulePath}/package.json`);
```

Comment:
to avoid loading it from the user's local `node_modules` if they're using globally installed sls

Reply:
Ok, I take it's to revert the effect of this hack: https://github.com/serverless/serverless/blob/856e0ec69e628d2929ac895a4979e960f7962f77/lib/classes/PluginManager.js#L148

As otherwise in normal case, `require` from global installation will never look into user local one.

Reply:
yup

Reply:
I'm a bit worried about number of hacks we apply in module loading, and some of them as I see introduce some side effects, which at some point may bite back.

I wonder whether it could be done less hacky.

1. What about adding: `module.paths.unshift(path.join(
      __dirname, '../../node_modules')` before this line: https://github.com/serverless/serverless/blob/856e0ec69e628d2929ac895a4979e960f7962f77/lib/classes/PluginManager.js#L155
So it's always `serverless/node_modules` that's searched first for plugins (and then service local directories)

2. Then I believe we do not need `filter` patch here: https://github.com/serverless/serverless/blob/856e0ec69e628d2929ac895a4979e960f7962f77/lib/classes/PluginManager.js#L156 (as if user mentions the plugin in `serverless.yml` it'll be double loaded and we end up here: https://github.com/serverless/serverless/blob/856e0ec69e628d2929ac895a4979e960f7962f77/lib/classes/PluginManager.js#L85

3. And here https://github.com/serverless/serverless/blob/856e0ec69e628d2929ac895a4979e960f7962f77/lib/classes/PluginManager.js#L163 we can rely on regular require as `@serverless/enterprise-plugin`

--- 

Other concern: Is this intentional to load SFE _after_ service plugins?
Shouldn't this plugin be loaded _after_ core plugins but before _service_ plugins which may interefere with SLS internals, or apply hooks that should otherwise be run after SFE hooks?